### ISSUE TITLE
[DocDB] The statistics `intentsdb_rocksdb_flush_write_bytes` and `rocksdb_flush_write_bytes` are increased after the flush, rather than during the flush

### ISSUE THREAD

**Issue by fritshoogland-yugabyte** [yugabyte/yugabyte-db#17503]
### Description

YugabyteDB version: 2.17.3
OS: Linux Alma 8.7

When looking at the tablet server write statistics:
- rocksdb_write_raw_blocks_micros_count (which indicates write IO requests for both regulardb and intents)
- intentsdb_rocksdb_flush_write_bytes (which indicates the amount of bytes written for intentdb memtable flush writing)
- rocksdb_flush_write_bytes (which indicates the amount of bytes written for regulardb memtable flush writing)

We can see the write IO requests happening for some time, and when write IO requests stop, the flush statistic seems to be increased with the total write size. 

This means that if you take flush bytes statistic and divide over time, it suggests the total flush happened at a single point in time, and therefore gives an impression of bandwidth (bytes per second) that is wrong, because in reality the write seemed to have been done over a longer period of time.

The solution would be to increase the flush write bytes statistic along with the write count statistic, so that the true bandwidth used can be seen.

The bandwidth used is a metric which most cloud virtual machines are bound by, which means that understand this statistic is crucial for IO tuning and general understanding of the performance properties of YugabyteDB.

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

**Comment by bmatican** [comment]
Indeed, it seems we update `WRITE_RAW_BLOCK_MICROS` in `BlockBasedTableBuilder::WriteRawBlock`, so we could similarly also update the size written to disk there. Unlike `FLUSH_WRITE_BYTES`, which we only update at the end of the FlushJob ...

Ironically, there's an unused constant in `src/yb/rocksdb/db/builder.cc`:
```
  // Reports the IOStats for flush for every following bytes.
  const size_t kReportFlushIOStatsEvery = 1048576;
```
cc @jmeehan16 @arybochkin @ttyusupov
